core/src/main/java/org/fao/geonet/kernel/Thesaurus.java

```diff
@@ -965,8 +983,17 @@ private void retrieveThesaurusInformation(Path thesaurusFile, String defaultTitl
                 this.defaultNamespace = DEFAULT_THESAURUS_NAMESPACE;
             }
 
-            Element dateEl = Xml.selectElement(thesaurusEl, "skos:ConceptScheme/dcterms:issued|skos:Collection/dc:date", theNSs);
+            Element issuedDateEl = Xml.selectElement(thesaurusEl, "skos:ConceptScheme/dcterms:issued|skos:Collection/dc:date", theNSs);
+            this.issuedDate = issuedDateEl==null? "": issuedDateEl.getText();
 
+            Element modifiedDateEl = Xml.selectElement(thesaurusEl, "skos:ConceptScheme/dcterms:modified", theNSs);
+            this.modifiedDate = modifiedDateEl==null? "": modifiedDateEl.getText();
+
+            Element createdDateEl = Xml.selectElement(thesaurusEl, "skos:ConceptScheme/dcterms:created", theNSs);
+            this.createdDate = createdDateEl==null? "": createdDateEl.getText();
+
+            // Default date
+            Element dateEl = Xml.selectElement(thesaurusEl, "skos:ConceptScheme/dcterms:issued|skos:Collection/dc:date", theNSs);
             Date thesaususDate = parseThesaurusDate(dateEl);
```

Comment:
For HNAP core subject thesaurus, the issued data would be 2004 which is an acceptable dc date.

However the parseThesaurusDate generates the following error message.

```
2023-04-20 23:18:34,908 ERROR [geonetwork.thesaurus-man] - Error parsing the thesaurus date value: 2004
2023-04-20T23:18:34.922650900Z   * with format: EEE MMM d HH:mm:ss z yyyy. Error is: Unparseable date: "2004"
2023-04-20T23:18:34.922661300Z   * with format: yyyy-MM-dd HH:mm:ss. Error is: Unparseable date: "2004"
2023-04-20T23:18:34.922667200Z   * with format: yyyy-MM-dd. Error is: Unparseable date: "2004"
2023-04-20T23:18:34.922673600Z Check thesaurus date in GC_Core_Subject.rdf
```

And since it cannot determine the date it will use a system date. 

In this case I think we should add the following below in the parseThesaurusDate so that it will end up using jan 1st 2004.

dfList.add(new SimpleDateFormat("yyyy"));

The following is also acceptable if we don't know the day of the month. So this could be added as well.
dfList.add(new SimpleDateFormat("yyyy-MM"));